app/src/main/kotlin/com/hippo/ehviewer/ui/main/AvatarIcon.kt

```diff
@@ -0,0 +1,201 @@
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import arrow.core.Either
+import arrow.core.Either.Companion.catch
+import arrow.core.Option
+import arrow.core.none
+import arrow.core.some
+import coil3.compose.AsyncImage
+import com.hippo.ehviewer.R
+import com.hippo.ehviewer.Settings
+import com.hippo.ehviewer.client.EhEngine
+import com.hippo.ehviewer.client.EhUtils
+import com.hippo.ehviewer.client.parser.HomeParser
+import com.hippo.ehviewer.collectAsState
+import com.hippo.ehviewer.ui.tools.DialogState
+import com.hippo.ehviewer.util.displayString
+import com.ramcosta.composedestinations.navigation.DestinationsNavigator
+import kotlin.time.Duration.Companion.seconds
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.flow.conflate
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.merge
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
+import moe.tarsin.coroutines.runSwallowingWithUI
+
+private val limitScope = CoroutineScope(Dispatchers.IO)
+private val refreshEvent = MutableSharedFlow<Unit>()
+private val invalidateEvent = MutableSharedFlow<Unit>()
+
+typealias Result = Option<Either<String, HomeParser.Result>>
+
+private val limitFlow: StateFlow<Result> = refreshEvent.conflate()
+    .onEach {
+        // If cached value is error, drop it when refresh
+        val isError = limitFlow.value.isSome { it.isLeft() }
+        if (isError) invalidateEvent.emit(Unit)
+    }
+    .map { catch { EhEngine.getImageLimits() }.mapLeft { e -> e.displayString() } }
+    .let { src -> merge(src.map { v -> v.some() }, invalidateEvent.map { none() }) }
+    .stateIn(limitScope, SharingStarted.Eagerly, none())
+    .apply {
+        limitScope.launch {
+            collectLatest {
+                // Consider cached value needs invalidate for 2 minutes
+                delay(120.seconds)
+                invalidateEvent.emit(Unit)
+            }
+        }
+    }
+
+context(CoroutineScope, DialogState, SnackbarHostState, DestinationsNavigator)
+@Composable
+fun AvatarIcon() {
+    val hasSignedIn by Settings.hasSignedIn.collectAsState()
+    if (hasSignedIn) {
+        val placeholder = stringResource(id = R.string.please_wait)
+        val resetImageLimitSucceed = stringResource(id = R.string.reset_limits_succeed)
+        val result by limitFlow.collectAsState()
+        IconButton(
+            onClick = {
+                launch {
+                    refreshEvent.emit(Unit)
```

Comment:
This is unnecessary now.

Reply:
This is necessary as user may want see updated value, but cache is not invalidated

Reply:
Then what's the cache for? It doesn't actually reduce network requests.

Reply:
For further pretty animation :)